ddpui/api/metrics_api.py

```
@@ -0,0 +1,326 @@
+
+
+@metrics_router.get("/{metric_id}/annotations/", response=List[AnnotationResponse])
+@has_permission(["can_view_charts"])
+def list_annotations(request, metric_id: int):
+    """Get all annotations for a metric"""
+    org = request.orguser.org
+
+    try:
+        metric = MetricDefinition.objects.get(id=metric_id, org=org)
+    except MetricDefinition.DoesNotExist:
+        raise HttpError(404, "Metric not found")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)